crates/forge/src/main.rs

```diff
@@ -177,52 +181,95 @@ fn test_workspace(args: TestArgs) -> Result<bool> {
         rt.spawn(async move {
             let mut all_failed_tests = vec![];
             for package in &packages {
-                let forge_config = config_from_scarb_for_package(&scarb_metadata, &package.id)?;
-                let (package_path, package_source_dir_path) =
-                    paths_for_package(&scarb_metadata, &package.id)?;
-                env::set_current_dir(package_path.clone())?;
-
-                let package_name = Arc::new(name_for_package(&scarb_metadata, &package.id)?);
-                let dependencies = dependencies_for_package(&scarb_metadata, &package.id)?;
-                let corelib_path = corelib_for_package(&scarb_metadata, &package.id)?;
-
-                let contracts = get_contracts_map(&scarb_metadata, &package.id).unwrap_or_default();
-
-                let runner_config = Arc::new(combine_configs(
-                    &workspace_root,
-                    args.exit_first,
-                    args.fuzzer_runs,
-                    args.fuzzer_seed,
-                    &forge_config,
-                ));
-
-                let runner_params = Arc::new(RunnerParams::new(
-                    corelib_path,
-                    contracts,
-                    env::vars().collect(),
-                    dependencies,
-                ));
-
-                let tests_file_summaries = run(
-                    &package_path,
-                    &package_name,
-                    &package_source_dir_path,
-                    &TestsFilter::from_flags(
-                        args.test_filter.clone(),
-                        args.exact,
-                        args.only_ignored,
-                        args.include_ignored,
-                        args.rerun_failed,
-                        workspace_root.join(CACHE_DIR),
-                    ),
-                    runner_config,
-                    runner_params,
-                    &forge_config.fork,
-                )
-                .await?;
-
-                let mut failed_tests = extract_failed_tests(tests_file_summaries);
-                all_failed_tests.append(&mut failed_tests);
+                if args.use_scarb_collector {
```

Comment:
Some of the stuff in repeats itself, I would draw them out of the conditional statement

Reply:
Most of them repeat, it was just the simples thing to do since we're removing it soon anyway.